kuwala/core/canvas/src/pages/DataSourcePreview.js

```diff
@@ -1,151 +1,364 @@
+                })
+            case 'bigquery':
+                return rawSchema.map((schema) => {
+                    return {
+                        ...schema,
+                        schema: schema.project,
+                        isOpen: false,
+                        categories: schema.datasets.map((data)=>{
+                            return {
+                                ...data,
+                                category: data.dataset,
+                                isOpen: false
+                            }
+                        })
+                    }
+                })
+            default:
+                return rawSchema
+        }
+    }
+
+    const renderTableDataPreview = (columns, rows) => {
+        return (
+            <table className={'bg-white'}>
+                <thead>
+                    <tr>
+                        {renderTableDataPreviewHead(columns)}
+                    </tr>
+                </thead>
+                <tbody>
+                        {renderTableDataPreviewBody(rows)}
+                </tbody>
+            </table>
+        )
+    }
+
+    const renderTableDataPreviewHead = (columns) => {
+        if(columns) {
+            return columns.map((e,i)=> (<th className={'sticky top-0 px-6 py-3 text-white bg-kuwala-green'}>{e}</th>))
+        } else {
+            return <></>
+        }
+    }
+
+    const renderTableDataPreviewBody = (rows) => {
+        if(rows) {
+            return rows.map((e,i) => (
+                <tr className={'bg-white border-2 text-center'}>
+                    {e.map((e,i)=> (<td className={'text-left px-4 py-2 border border-kuwala-green'}>{
+                        JSON.stringify(e).trim()
+                    }</td>))}
+                </tr>
+            ))
+        } else {
+            return <></>
+        }
+    }
+
+    const tableSelectionOnlick = async (addressString) => {
+        setSelectedTable(addressString)
+        setIsTableDataPreviewLoading(true)
+        const params = generateParamsByDataSourceType(selectedSource.data_catalog_item_id, addressString)
+        const res = await getTablePreview({
+            id: dataIndex,
+            params
+        });
+
+        if(res.status === 200) {
+            setTableDataPreview(res.data)
+        }
+        setIsTableDataPreviewLoading(false)
+    }
+
+    const generateParamsByDataSourceType = (type, addressString) => {
+        const arr = addressString.split('@')
+        switch (type){
+            case "postgres":
+                return {
+                    schema_name: arr[0],
+                    table_name: arr[2],
+                    limit_columns: 100,
+                    limit_rows: 100,
+                }
+            case "bigquery":
+                return {
+                    project_name: arr[0],
+                    dataset_name: arr[1],
+                    table_name: arr[2],
+                    limit_columns: 100,
+                    limit_rows: 100,
```

Comment:
Column limit should be 200 and row limit 300 (see non-functional requirements in issues).